src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -551,7 +551,8 @@ TransactionManager::get_extents_if_live(
   return cache->get_extent_if_cached(t, paddr, type
   ).si_then([=, this, &t](auto extent)
 	    -> get_extents_if_live_ret {
-    if (extent && extent->get_length() == len) {
```

Comment:
This was introduced in commit https://github.com/ceph/ceph/pull/45519/commits/ec90bcbfec5a26563ae2b9abc1d1d31d901c0a27, which was to address the case "when the target extent may have been split into multiple pieces by other transaction".

From previous discussions with @zhscn, it should be possible that the extent has been splitted in a parallel (mutate) transaction so that `extent->get_length() != len`, in that case, we should go to the else case to reload it from lba-tree.

Reply:
It's been a while since the split PR, so I might forget some details. The primary purpose of this length predicate was to verify if the cached backref entries expire between the time when `BackrefManager::get_cached_backref_entries_in_range()` is called and when `TransactionManager::get_extents_if_live()` is invoked within the `SegmentCleaner::do_reclaim_space()`.

The real call stack is:
```cpp
SegmentCleaner::do_reclaim_space() {
    //...
    auto cached_backref_entries =
          backref_manager.get_cached_backref_entries_in_range(
            reclaim_state->start_pos, reclaim_state->end_pos);
    //...
    return seastar::do_with() {
        //...
        return trans_intr::parallel_for_each() {
            //...
            return extent_callback->get_extents_if_live();
        }
    }
}
```

It seems that these two continuations(do_with and parallel_for_each) will not be scheduled out by Seastar. So the retrieved backref entries should always be the latest so that we could change this predicate to assertion.

Reply:
I think we should not rely on the atomiciy across multiple continuations. And it doesn't seem a huge effort with the additional condition `extent->get_length() == len` to make sure we are correct, right?